parser/expression_parsing.go

```diff
@@ -313,7 +314,20 @@ func (p *Parser) parseCallExpression(receiver ast.Expression) ast.Expression {
 		// current token is identifier (method name)
 		exp = &ast.CallExpression{Token: p.curToken, Receiver: receiver, Method: m}
 		exp.Arguments = p.parseCallArguments()
-	} else { // call expression has a receiver like: p.foo
+	} else if p.curTokenIs(token.Ident) {
+
```

Comment:
Make sure you refactor this part and above section since they have only a few differences